src/universal/modules/welcome/containers/Welcome/Welcome.js

```diff
@@ -32,16 +49,27 @@ const mapStateToProps = state => ({
 export default class WelcomeContainer extends Component {
   static propTypes = {
     dispatch: PropTypes.func,
+    invitees: PropTypes.array,
+    inviteesRaw: PropTypes.string,
+    preferredName: PropTypes.string,
+    teamName: PropTypes.string,
     welcome: PropTypes.shape({
-      preferredName: PropTypes.string,
-      teamName: PropTypes.string,
       teamId: PropTypes.string,
       teamMemberId: PropTypes.string
     })
   };
 
+  constructor(props) {
+    super(props);
+    this.onPreferredNameSubmit = this.onPreferredNameSubmit.bind(this);
```

Comment:
ahhhhh. good call. Would you mind if i refactored those methods into their respective stateless components? The whole only-expose-data-as-high-as-you-need-it thing is started to make me think i'm doing it wrong.